### ISSUE TITLE
RoboFile.php is missing for gitlab-ci

### ISSUE THREAD

**Issue by visabhishek** [Lullabot/drupal8ci#31]
Getting Error while trying to run the "build"

Please help me to Fix this

![screenshot from 2018-05-18 18-23-58](https://user-images.githubusercontent.com/2493261/40235700-ac9a0baa-5ac8-11e8-9ede-474e130e6a5e.png)

**Comment by deviantintegral** [comment]
Can you manually copy in the file from https://github.com/Lullabot/drupal8ci/blob/master/dist/circleci/.circleci/RoboFile.php to fix it for now?

**Comment by visabhishek** [comment]
Still, I am getting an error " There are no commands defined in the "install" namespace.
![screenshot from 2018-05-22 18-30-59](https://user-images.githubusercontent.com/2493261/40363881-72026c84-5dee-11e8-94df-e6cc015fa7c9.png)

**Comment by visabhishek** [comment]
Still, I am getting this issue.
![screenshot from 2018-05-24 13-54-49](https://user-images.githubusercontent.com/2493261/40473734-a307d89c-5f5a-11e8-88d7-afba555f0d5d.png)

My robofile is : https://github.com/visabhishek/drupal8ci/blob/master/dist/gitlabci/.gitlab-ci/RoboFile.php

Am I missing anything?

**Comment by juampynr** [comment]
I am sorry @visabhishek but the GitLab integration is not in feature parity with the CircleCI and Travis CI ones. We have received a few requests about this so we have planned to work on it. I personally will spend a few hours this week and post my progress in a pull request so keep an eye on the repository.

**Comment by devkinetic** [comment]
Any progress on this?

**Comment by juampynr** [comment]
Hi @devkinetic! As far as I can remember, it was @q0rban who did the initial effort to support GitLab. I still haven't got the chance to work in it so I don't know how it is supposed to be integrated.

As for the images, Travis uses the wodby images (I think that I was just experimenting with them when I did the integration) but [CircleCI uses juampynr/drupal8ci](https://github.com/Lullabot/drupal8ci/blob/master/dist/circleci/.circleci/config.yml#L14), which adds a few useful tools for Drupal 8 projects. If I was to decide, I would use juampynr/drupal8ci as the wodby one may not have everything that you need for CI.

**Comment by devkinetic** [comment]
It looks like what was built into the RoboFile was mostly migrated into the gitlab-ci file. For the implementation, there were two components missing that are called out to:

- `robo setup:drupal || true`
- `robo install:dependencies`

I'm assuming that @q0rban paired down the robofile to just particular tasks.

**Comment by q0rban** [comment]
@devkinetic @juampynr when I first added the gitlab support, it was prior to a refactor that split things out into separate directories. At that time, I adapted the RoboFile.php that CircleCI used to work with GitLab. So perhaps try to copy that Robofile into place and see if that works. 

You can view my work here: https://github.com/Lullabot/drupal8ci/pull/14/files

**Comment by devkinetic** [comment]
I got much farther from that starting point, thank you @q0rban . I was able to flesh out the migration to the new robofile, but i've hit a snag here:
```
 [Composer\Install] Done in 59.89s
 [Exec] Running dockerize -wait tcp://localhost:3306 -timeout 1m
2019/05/03 04:59:03 Waiting for: tcp://localhost:3306
2019/05/03 04:59:03 Problem with dial: dial tcp [::1]:3306: getsockopt: connection refused. Sleeping 1s
........
2019/05/03 05:00:03 Timeout after 1m0s waiting on dependencies to become available: [tcp://localhost:3306]
 [Exec]  Exit code 1  Time 01:00
ERROR: Job failed: exit code 1
```
This is my current gitlab-ci file, it's currently missing the caching pieces, but otherwise was getting close to feature parity based on the circleci one I was following from master. I'm not really sure why the sql port doesn't become available.

```
stages:
  - test_and_lint

run-unit-kernel-tests:
  stage: test_and_lint
  extends: .unit_kernel_tests

#run-behat-tests:
#  stage: test_and_lint
#  extends: .behat-tests

run-code-sniffer:
  stage: test_and_lint
  extends: .code-sniffer

run-code-coverage:
  stage: test_and_lint
  extends: .code-coverage

.defaults:
  image: juampynr/drupal8ci:latest
  services:
    - mariadb:10.3
#    - selenium/standalone-chrome-debug:3.7.1-beryllium
  variables:
    MYSQL_ALLOW_EMPTY_PASSWORD: 1
  before_script:
    - cp .gitlab-ci/RoboFile.php .
    - mkdir -p ~/.ssh
    - chmod 700 ~/.ssh
    - 'which ssh-agent || ( apt-get update -y && apt-get install openssh-client -y )'
    - eval $(ssh-agent -s)
    - ssh-add <(echo "$SSH_PRIVATE_KEY")
    - '[[ -f /.dockerenv ]] && echo -e "Host *\n\tStrictHostKeyChecking no\n\n" > ~/.ssh/config'
    - ln -s $CI_PROJECT_DIR/web /var/www/html/web
    - apache2-foreground&
  artifacts:
    paths:
      - vendor
      - bin
      - web

.unit_kernel_tests:
  extends: .defaults
  script:
    - cp .gitlab-ci/RoboFile.php .
    - robo job:run-unit-tests
  artifacts:
    paths:
      - artifacts

#.behat-tests:
#  extends: .defaults
#  script:
#    - cp .gitlab-ci/RoboFile.php .
#    - robo job:run-behat-tests
#  artifacts:
#    paths:
#      - artifacts

.code-sniffer:
  extends: .defaults
  script:
    - cp .gitlab-ci/RoboFile.php .
    - robo job:heck-coding-standards
  artifacts:
    paths:
      - artifacts

.code-coverage:
  extends: .defaults
  script:
    - cp .gitlab-ci/RoboFile.php .
    - robo job:generate-coverage-report
  artifacts:
    paths:
      - artifacts```

**Comment by devkinetic** [comment]
I've gotten quite far into a refactor for Gitlab. At this point I'm running into issues with `drush site-install':

`Fatal error: Maximum execution time of 60 seconds exceeded in /builds/****/vendor/symfony/dependency-injection/Definition.php on line 114`

I tried bumping up everything I could think of with a new Dockerfile, only to keep getting the same error.

```
FROM juampynr/drupal8ci

RUN echo 'memory_limit = -1' > /usr/local/etc/php/php.ini
RUN echo 'max_execution_time = 1200' > /usr/local/etc/php/php-cli.ini
RUN echo 'max_execution_time = 1200' > /usr/local/etc/php/php.ini
RUN echo 'max_input_time = 1200' > /usr/local/etc/php/php-cli.ini
RUN echo 'max_input_time = 1200' > /usr/local/etc/php/php.ini
```

**Comment by juampynr** [comment]
@devkinetic your snippet above looks fine. I would find a way to jump into the container to verify with`php -i   | grep max_execution_time` is getting the value that you have set.

**Comment by juampynr** [comment]
I have reworked the GitLab installer and added a RoboFile.php to it. Closing. Please try the installer again and open a new ticket if you find any issues with it.

Thanks for your feedback @visabhishek and @devkinetic!

**Comment by devkinetic** [comment]
I'll take another look at this! Thanks!
